src/components/CreateNewBudgetSheet.js

```diff
@@ -0,0 +1,177 @@
+import { useState } from 'react';
+import firebase from '../firebase';
+import { getDatabase, ref, push } from 'firebase/database';
+import { useNavigate } from 'react-router-dom';
+import WarningModal from './WarningModal';
+
+const NewBudgetSheet = (props) => {
+  const database = getDatabase(firebase);
+  const dbRef = ref(database);
+
+  const [name, setName] = useState('');
+  const [budgetTotal, setBudgetTotal] = useState('');
+  const [categoriesInput, setCategoriesInput] = useState('');
+  const [categoriesArray, setCategoriesArray] = useState([]);
+  const [isWarning, setIsWarning] = useState(false);
+
+  // Forms
+  const handleNameInputChange = (e) => {
+    setName(e.target.value);
+  };
+
+  const handleBudgetInputChange = (e) => {
+    setBudgetTotal(e.target.value);
+  };
+
+  let navigate = useNavigate();
+
+  const handleSubmit = (e, budgetAmount, nameOfSheet, categories) => {
+    e.preventDefault();
+    if (categories.length === 0) {
+      setIsWarning(true);
```

Comment:
If I understand this, you're using a state variable to handle the warning for an incorrectly filled out new budget form. Is that necessary? Does it need to store this information between renders, and update something when this information changes? I assume you're using this state to show the warning popup when it does re-render—which is neat. But maybe it would make sense to do it differently?